go/ql/lib/semmle/go/frameworks/GinCors.qll

```diff
@@ -0,0 +1,115 @@
+/**
+ * Provides classes for modeling the `github.com/gin-contrib/cors` package.
+ */
+
+import go
+
+/**
+ * Provides classes for modeling the `github.com/gin-contrib/cors` package.
+ */
+module GinCors {
+  /** Gets the package name `github.com/gin-gonic/gin`. */
+  string packagePath() { result = package("github.com/gin-contrib/cors", "") }
+
+  /**
+   * A new function create a new gin Handler that passed to gin as middleware
+   */
+  class New extends Function {
+    New() { exists(Function f | f.hasQualifiedName(packagePath(), "New") | this = f) }
+  }
+
+  /**
+   * A write to the value of Access-Control-Allow-Credentials header
+   */
+  class AllowCredentialsWrite extends DataFlow::ExprNode {
+    DataFlow::Node base;
+    GinConfig gc;
+
+    AllowCredentialsWrite() {
+      exists(Field f, Write w |
```

Comment:
```suggestion
      exists(Field f, Write w, DataFlow::Node base |
```
and get rid of it as a field in this class (because it's never used for anything).